extensions/ql-vscode/src/query-results.ts

```diff
@@ -253,13 +270,15 @@ export class FullQueryInfo {
   constructor(
     public readonly initialInfo: InitialQueryInfo,
     config: QueryHistoryConfig,
-    private readonly source?: CancellationTokenSource
+    private source?: CancellationTokenSource // used to cancel in progress queries
   ) {
     this.setConfig(config);
   }
 
   cancel() {
     this.source?.cancel();
+    // query is no longer in progress, can delete the cancellation token source
+    delete this.source;
```

Comment:
Why do we need to do this?

Reply:
The `source` object is complex and contains workspace state. I'm doing it to make sure workspace resources aren't leaked. I realized that this object is a `Disposable` as well, so I'll add a call to `dispose`.

Slightly unrelated, this object can't be serialized to JSON because it's recursive, so it also needs to be deleted before serialization.

Reply:
That sounds fun 😱